src/CMakeLists.txt

```diff
@@ -0,0 +1,159 @@
+    ${PROTO_GEN_DIR}/gen/google/api/annotations.pb.h
+    ${PROTO_GEN_DIR}/gen/google/api/http.pb.cc
+    ${PROTO_GEN_DIR}/gen/google/api/http.pb.h
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.grpc.pb.cc
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.grpc.pb.h
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.pb.cc
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.pb.h
+
+    # These are needed for `echo` example, which is awkward. These can
+    # probably be removed if that example is pulled.
+    ${PROTO_GEN_DIR}/gen/proto/rpc/examples/echo/v1/echo.grpc.pb.cc
+    ${PROTO_GEN_DIR}/gen/proto/rpc/examples/echo/v1/echo.grpc.pb.h
+    ${PROTO_GEN_DIR}/gen/proto/rpc/examples/echo/v1/echo.pb.cc
+    ${PROTO_GEN_DIR}/gen/proto/rpc/examples/echo/v1/echo.pb.h
+
+    COMMAND buf generate buf.build/viamrobotics/goutils --template buf.gen.yaml
+    COMMAND buf generate buf.build/googleapis/googleapis --template buf.gen.yaml --path google/rpc --path google/api
+    COMMAND buf generate buf.build/viamrobotics/api --template buf.gen.yaml --path common,component,robot,service,module
+    MAIN_DEPENDENCY buf.gen.yaml
+  )
+
+  add_custom_target(
+    update-static-protos
+    # This must be one of the files listed in `add_custom_command` above,
+    # but it doesn't matter which one, we just need to have a dependency edge
+    # into the files produced by that command.
+    #
+    # TODO: If we DRY as suggested above, this could just be the first list item.
+    #
+    DEPENDS gen/common/v1/common.grpc.pb.cc
+  )
+
+  add_custom_command(
+    TARGET update-static-protos
+    COMMAND ${CMAKE_COMMAND} -E copy_directory ${PROTO_GEN_DIR}/gen ${CMAKE_SOURCE_DIR}/src/gen
+  )
+endif(VIAMCPPSDK_USE_DYNAMIC_PROTOS)
+
+# I dislike exporting values via `CACHE` but for now we need this
+# state over in the `examples` directory. Hopefully when we do some
+# physical refactoring that can be eliminated.
+set(PROTO_GEN_DIR ${PROTO_GEN_DIR} CACHE INTERNAL "")
+
+
+# Declare the viamcpp library, and attach sources,
+# headers for installation
+add_library(viamcpp)
+
+target_sources(viamcpp
+  PRIVATE
+    ${PROTO_GEN_DIR}/gen/common/v1/common.grpc.pb.cc
+    ${PROTO_GEN_DIR}/gen/common/v1/common.pb.cc
+    ${PROTO_GEN_DIR}/gen/google/api/annotations.pb.cc
+    ${PROTO_GEN_DIR}/gen/google/api/http.pb.cc
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.pb.cc
+    ${PROTO_GEN_DIR}/gen/robot/v1/robot.grpc.pb.cc
+    common/proto_type.cpp
+    common/utils.cpp
+    components/component_base.cpp
+    components/resource_manager.cpp
+    registry/registry.cpp
+    robot/client.cpp
+    rpc/dial.cpp
+  PUBLIC FILE_SET viamcpp_public_includes TYPE HEADERS
+    FILES
+      common/proto_type.hpp
+      common/utils.hpp
+      components/component_base.hpp
+      components/resource_manager.hpp
+      components/service_base.hpp
+      registry/registry.hpp
+      robot/client.hpp
+      robot/service.hpp
+      rpc/dial.hpp
+  PUBLIC FILE_SET viamcpp_public_pb_includes TYPE HEADERS
```

Comment:
I'd like to know if we intend for the generated protos to form part of the public API of the library, or if we want to fully wrap them. My instinct is to say that they shouldn't be exposed as part of the SDK API, but I may be being overly conservative.

Reply:
The proto-generated stuff is already "published" in its protobuf form elsewhere, so my opinion would agree with yours, that the SDK should fully wrap things. Though in practice, I expect some small bits of the generated stuff may get raw exposure (such as timestamp formatting) just as it does in Go.